Utility/PublicObjectView.cs

```diff
@@ -0,0 +1,34 @@
+﻿//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+namespace Microsoft.PackageManagement.NuGetProvider
+{
+    /// <summary>
+    /// Allows internal object to be used in "public" APIs
+    /// </summary>
+    public class PublicObjectView
```

Comment:
what's the purpose of this class?

Reply:
Some APIs needed to made public but a parameter is internal. Instead of changing internal -> public and causing cascading changes, this is simpler, if a bit ugly.

---
In reply to: [154462656](https://github.com/OneGet/NuGetProvider/pull/25#discussion_r154462656) [](ancestors = 154462656)